connect/runtime/src/main/java/org/apache/kafka/connect/util/TopicAdmin.java

```diff
@@ -375,6 +383,152 @@ public boolean createTopic(NewTopic topic) {
         return existingTopics;
     }
 
+    /**
+     * Verify the named topic uses only compaction for the cleanup policy.
+     *
+     * @param topic             the name of the topic
+     * @param workerTopicConfig the name of the worker configuration that specifies the topic name
+     * @return true if the admin client could be used to verify the topic setting, or false if
+     *         the verification could not be performed, likely because the admin client principal
+     *         did not have the required permissions or because the broker was older than 0.11.0.0
+     * @throws ConfigException if the actual topic setting did not match the required setting
+     */
+    public boolean verifyTopicCleanupPolicyOnlyCompact(String topic, String workerTopicConfig,
+            String topicPurpose) {
+        Set<String> cleanupPolicies = topicCleanupPolicy(topic);
+        if (cleanupPolicies == null || cleanupPolicies.isEmpty()) {
+            log.debug("Unable to use admin client to verify the cleanup policy of '{}' "
+                      + "topic is '{}', either because the broker is an older "
+                      + "version or because the Kafka principal used for Connect "
+                      + "internal topics does not have the required permission to "
+                      + "describe topic configurations.", topic, TopicConfig.CLEANUP_POLICY_COMPACT);
+            return false;
+        }
+        String cleanupPolicyStr = String.join(",", cleanupPolicies);
+        log.debug("Found cleanup policy for '{}' topic is '{}'", topic, cleanupPolicyStr);
+        Set<String> expectedPolicies = Collections.singleton(TopicConfig.CLEANUP_POLICY_COMPACT);
+        String expectedPolicyStr = String.join(",", expectedPolicies);
+        if (cleanupPolicies != null && !cleanupPolicies.equals(expectedPolicies)) {
+            String msg = String.format("Topic '%s' supplied via the '%s' property is required "
+                                       + "to have '%s=%s' to guarantee consistency and durability of "
+                                       + "%s, but found '%s'. "
+                                       + "Correct the topic before restarting Connect.",
+                    topic, workerTopicConfig, TopicConfig.CLEANUP_POLICY_CONFIG, expectedPolicyStr,
+                    topicPurpose, cleanupPolicyStr);
+            throw new ConfigException(msg);
+        }
+        return true;
+    }
+
+    /**
+     * Get the cleanup policy for a topic.
+     *
+     * @param topic the name of the topic
+     * @return the set of cleanup policies set for the topic; may be empty if the topic exists
```

Comment:
I wonder if distinguishing like that, with `null` and `empty` pays off. 
Why not return an empty collection in both cases and simplify the checks on the return values of this method?

This method doesn't seem to be the one to use when somebody wants to determine whether a topic exists or not.